http-clients/netty-nio-client/src/test/java/software/amazon/awssdk/http/nio/netty/fault/SdkTestHttpContentPublisher.java

```diff
@@ -0,0 +1,56 @@
+package software.amazon.awssdk.http.nio.netty.fault;
+
+import java.nio.ByteBuffer;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+import software.amazon.awssdk.http.async.SdkHttpContentPublisher;
+
+public class SdkTestHttpContentPublisher implements SdkHttpContentPublisher {
```

Comment:
Can we use the existing `SimpleHttpContentPublisher` or something from RxJava?

Reply:
`SimpleHttpContentPublisher` is in core and netty didnot have core dependency, I did not find anything from RxJava, will keep looking out for it.

Reply:
Added SimpleHttpContentPublisher in `http-client-tests` since adding in `test-utils` is casuing cyclic dependency.